### ISSUE TITLE
Issue with unmodified display name 

### ISSUE THREAD

**Issue by CredibleOpossum** [serenity-rs/serenity#2542]
Using [display_name](https://docs.rs/serenity/latest/serenity/model/guild/struct.Member.html#method.display_name) seems to incorrectly (?) use the name of the discord account and not the display name when using a unmodified discord name.

![image](https://github.com/serenity-rs/serenity/assets/60998830/10b9445d-f06b-4cf1-8804-c34f32274bf3)
this name will return `lexvengeance`

![image](https://github.com/serenity-rs/serenity/assets/60998830/497bef23-eebc-46bb-893e-45f5a4d07295)
this name will return `+[------->++<]>++.-..+` (I removed a single dot)

Let me know if I am confused or using the wrong tool for the job

**Comment by jamesbt365** [comment]
Now discord makes shows this a bit weird so I'll try my best to explain and hope I'm right.

First off `current` doesn't have support for discord display names yet, the display_name on `Member` is what name is actually *displayed*, not the display name of the user. Display names are supported in [next](https://serenity-rs.github.io/serenity/next/serenity/model/prelude/struct.User.html#structfield.global_name). I'll call these global names from now on.

I'm going to assume that`+[------->++<]>++.-..+.` is the global name of the user, which as current doesn't support will just account for the nickname and username. As the nickname isn't set it will just use the username.

tldr, `next` has support for global names, the property you assume to be the right one just shows the name which is displayed, but `current` doesn't have any global names yet. (no idea if it will either) See #2500 

Assuming I'm right maybe some clarity could be added to the documentation?


God this is such a mindfuck to explain and I'm not even sure I'm 100% right either.

**Comment by CredibleOpossum** [comment]
Using the "next" branch fixed it (minimal modifications)

let me describe the problem because as I can't understand with the different

**Comment by CredibleOpossum** [comment]
Works using the "next" branch. (minimal code changes)

To be clearer on the problem (because I'm really confused on the terminology)

![image](https://github.com/serenity-rs/serenity/assets/60998830/c734089f-d7b3-43a4-a548-4025f7f8d5f5)

The red name seems to be used as there's no nickname set for the guild. Even though the nickname isn't set, I would expect the green one to be used. 

For some reason now on discord these are 2 separate names, red one is unique while green doesn't have to be.

I'd assume this is a bug in the current branch as the behavior seems to match what I expect when I switch.

**Comment by jamesbt365** [comment]
> I'd assume this is a bug in the current branch as the behavior seems to match what I expect when I switch.

Not really a "bug" per se, but a missing feature (global names).
